app/helpers/generic_file_helper.rb

```diff
@@ -39,7 +39,19 @@ def download_path(*args)
       gf = GenericFile.find(item.id)
       return gf.doi_url if gf.doi_url.present?
     end
-    return sufia.download_path(*args)
+    if args[1].nil?
+      return "/files/" + item.id + "/" + URI::encode(item.label) unless item.label.nil?
+      return "/files/" + item.id + "/nofile"
```

Comment:
These would be good conditions to write specs for. Why is the 'nofile' important?